google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreOpenTelemetryOptions.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import io.opentelemetry.api.OpenTelemetry;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class FirestoreOpenTelemetryOptions {
+  private final boolean enabled;
+
+  @Nullable private final OpenTelemetry openTelemetry;
+
+  FirestoreOpenTelemetryOptions(Builder builder) {
+    this.enabled = builder.enabled;
+    this.openTelemetry = builder.openTelemetry;
+  }
+
+  public boolean getEnabled() {
+    return enabled;
+  }
+
+  public OpenTelemetry getOpenTelemetry() {
+    return openTelemetry;
+  }
+
+  @Nonnull
+  public FirestoreOpenTelemetryOptions.Builder toBuilder() {
+    return new FirestoreOpenTelemetryOptions.Builder(this);
+  }
+
+  @Nonnull
+  public static FirestoreOpenTelemetryOptions.Builder newBuilder() {
+    return new FirestoreOpenTelemetryOptions.Builder();
+  }
+
+  public static class Builder {
+
+    private boolean enabled;
+
+    @Nullable private OpenTelemetry openTelemetry;
+
+    private Builder() {
+      enabled = false;
```

Comment:
why do we need these if it's essentially unused (unless I'm missing something)?

Reply:
sorry, it's not clear here, but this [will get used in the next PR](https://github.com/googleapis/java-firestore/pull/1533/files#diff-af6785fb3ddb14c6ee2fb8a813c9309fe4a81beefe5d37ab33fca7faedc28c2eR65). I tried to break down the large prototype PR to smaller PRs as much as it logically made sense.